src/components/content/order/formElements/TextInput.tsx

```diff
@@ -3,23 +3,61 @@
  * SPDX-FileCopyrightText: Huawei Inc.
  */
 
-import { Form, Input, Select, Tooltip } from 'antd';
+import { AutoComplete, Form, Input, Select, Tooltip } from 'antd';
 import { EyeInvisibleOutlined, EyeTwoTone, InfoCircleOutlined } from '@ant-design/icons';
-import { DeployVariable } from '../../../../xpanse-api/generated';
-import React, { ChangeEvent } from 'react';
+import { DeployRequest, DeployVariable } from '../../../../xpanse-api/generated';
+import React, { ChangeEvent, useEffect, useState } from 'react';
 import { Rule } from 'rc-field-form/lib/interface';
 import { useOrderFormStore } from '../store/OrderFormStore';
 import { DeployParam } from '../types/DeployParam';
 import { DeployVariableSchema } from '../types/DeployVariableSchema';
+import useAutoFillDeployVariableQuery from '../create/useAutoFillDeployVariableQuery';
 
-export function TextInput({ item }: { item: DeployParam }): React.JSX.Element {
+export function TextInput({
+    item,
+    csp,
+    region,
+}: {
+    item: DeployParam;
+    csp: DeployRequest.csp;
+    region: string;
+}): React.JSX.Element {
+    const [variableNameList, setVariableNameList] = useState<string[]>([]);
+    const [options, setOptions] = useState<{ value: string }[]>([]);

```

Comment:
this is not a state. We can use a normal variable. 
State must be only those values which must capture user actions.